scipy/io/netcdf.py

```diff
@@ -509,7 +509,12 @@ def _write_var_data(self, name):
             # Handle rec vars with shape[0] < nrecs.
             if self._recs > len(var.data):
                 shape = (self._recs,) + var.data.shape[1:]
-                var.data.resize(shape)
+                # Resize in-place does not always work since 
+                # the array might not be single-segment                              
+                try:
+                    var.data.resize(shape)
+                except ValueError:
+                    var.data = np.resize(var.data, shape).astype(var.data.dtype)
```

Comment:
Suggest writing it as `var.__dict__['data'] = ...` so that you don't need to add a special case in `__setattr__`.